### DISCUSSION TITLE
Add required reviewers in Azure DevOps PRs

### DISCUSSION THREAD

**Discussion by sch1ldkr0ete** [renovatebot/renovate#17659] [Request Help] [answered]
Is there the possibility to mark reviewers as required when creating Azure DevOps PRs? After a quick look at the code it does not seem like it is possible. In my opinion this would be very helpful. More than happy to contribute if necessary.

**Comment by rarkins** [comment]
Here is the current capability: https://github.com/renovatebot/renovate/blob/997f7f1a426ad31d45a65bada653d9b41b411049/lib/modules/platform/azure/index.ts#L845-L870

How does your requirement differ?

**Reply by sch1ldkr0ete** [reply]
The `isRequired` option of the `createPullRequestReviewer` method is not used. So it is not possible to declare a reviewer as required for Azure DevOps. (It seems like the default value is set to `false`.) It would be nice to have the possibility to do that. For example with adding a `req:` before the name of the reviewer.

**Reply by rarkins** [reply]
Is it one of the available options in `azureApiGit.createPullRequestReviewer()`?

**Reply by sch1ldkr0ete** [reply]
Yes, it is available via [IdentityRefWithVote](https://github.com/microsoft/azure-devops-node-api/blob/master/api/interfaces/GitInterfaces.ts#L2810):
https://github.com/microsoft/azure-devops-node-api/blob/master/api/GitApi.ts#L4014

**Reply by sch1ldkr0ete** [reply]
To indicate that a reviewer is a required reviewer I was thinking of a `req:` before the user or team name. So the implementation would look something like [this](https://github.com/sch1ldkr0ete/renovate/commit/7b619c0af652f75792892c2129df4360f07478f0). What do you think?

**Reply by rarkins** [reply]
Maybe spell it all out as "required:" so that it's easier for people reading the config to understand without needing to read the renovate docs?

**Comment by sch1ldkr0ete** [comment, marked as answer]
Feature is now available via [18111](https://github.com/renovatebot/renovate/pull/18111).
